app/models/view/scheduling/Periodicity.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2019 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package models.view.scheduling;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.collect.Maps;
+
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+/**
+ * {@code Periodicity} represents a repetition strategy for a {@link Schedule}.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+public enum Periodicity {
+    /**
+     * Every minute.
+     */
+    MINUTELY("Minutely", ChronoUnit.MINUTES),
+    /**
+     * Every hour.
+     */
+    HOURLY("Hourly", ChronoUnit.HOURS),
+    /**
+     * Every day.
+     */
+    DAILY("Daily", ChronoUnit.DAYS);
+
+    private static final Map<String, Periodicity> BY_NAME = Maps.newHashMap();
+
+    static {
+        for (Periodicity p : Periodicity.values()) {
+            BY_NAME.put(p._name, p);
+        }
+    }
+
+    private final String _name;
```

Comment:
Java enum's already have a "name"; we should avoid redefining that concept.

Reply:
Also, enums in the internal package are frequently used in the view model and storage model directly because they are enumerations (e.g. domain identifiers w/o state). Such a strategy with this enum may simplify the implementation considerably.

Reply:
I wasn't aware of that convention - I had actually been thinking this was a lot of overhead since this class already essentially exists. I'll try refactoring this to use that then.